File: 2022/day-1/main.lua
f = io.open("./input", "r");

local elves = {};
local i = 1;
elves[i] = 0
while true do
  local line = f:read();
  if line == nil then break end

  if #line == 0 then
    i = i + 1;
    elves[i] = 0;
  else
    elves[i] = elves[i] + tonumber(line);
  end
end

local max = 0;
for _, value in pairs(elves) do
  max = math.max(max, value)
end

print(max)

File: 2022/day-1/src/main.rs
const INPUT: &str = include_str!("../input");
fn main() {
    let mut elves = INPUT.lines().fold(vec![0u32], |mut acm, elm| {
        if elm.is_empty() {
            acm.push(0);
        } else if let Some(elf) = acm.last_mut() {
            let val = elm.parse::<u32>().unwrap();
            *elf += val;
        }
        acm
    });
    let max = elves.iter().max().unwrap();
    println!("{}", max);

    elves.sort();
    let sum: u32 = elves.iter().rev().take(3).sum::<u32>();
    println!("{}", sum)
}
